### PR TITLE
ENH: Add asv_bench_memray

### PR DESCRIPTION
**Pull request by HaoZeke**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

@conda-forge-admin, please ping conda-forge/help-python

### CODE REVIEW

**Comment by HaoZeke** [comment]
I'm the submitting author and the maintainer listed in the recipe, I can confirm I am willing to maintain this submission.

**Comment by HaoZeke** [comment]
@conda-forge/help-python, ready for review!

**Comment by HaoZeke** [comment]
@conda-forge/help-python, ready for review!

File: recipes/asv_bench_memray/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/asv_bench_memray-{{ version }}.tar.gz
+  sha256: 654a5bb989e15d2a6b46d85349b492f83605ae1e89d95f76f8f5943d5723cb8f
+
+build:
+  skip: true  # [win]
```
Comment (ocefpaf) [inline recipes/asv_bench_memray/meta.yaml:13]: Looks like this on could be noarch.

It is OK for noarch package to fails on Windows with a missing dependency. We prefer noarch b/c it is faster to build and the package will be ready when the missing dependency is available on Windows.

Comment (HaoZeke) [reply recipes/asv_bench_memray/meta.yaml]: Makes sense, updated, sorry for the long delay!!

**Comment by HaoZeke** [comment]
@conda-forge/help-python, ready for review!

**Review by xhochy — APPROVED** [review]

**Pull request merged**